### PR TITLE
Migrate over to new terraform SDK

### PR DESCRIPTION
**Pull request by masutaka**
Per https://www.terraform.io/docs/extend/plugin-sdk.html.

It seemed to be a seamless upgrade.

* [x] TODO: When PR #15 is merged, I'll rebase this branch

### CODE REVIEW

**Comment by kristofferahl** [comment]
@masutaka Let me know when the PR is rebased and I'll have a look. Great work my friend!

**Comment by masutaka** [comment]
@kristofferahl Please review!

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,30 @@ module github.com/kristofferahl/terraform-provider-healthchecksio
...
+	golang.org/x/crypto v0.0.0-20191002192127-34f69633bfdc // indirect
+	golang.org/x/exp v0.0.0-20191002040644-a1355ae1e2c3 // indirect
+	golang.org/x/lint v0.0.0-20190930215403-16217165b5de // indirect
+	golang.org/x/net v0.0.0-20191007182048-72f939374954 // indirect
+	golang.org/x/sys v0.0.0-20191008105621-543471e840be // indirect
+	golang.org/x/tools v0.0.0-20191007185444-6536af71d98a // indirect
+	google.golang.org/api v0.10.0 // indirect
+	google.golang.org/appengine v1.6.5 // indirect
+	google.golang.org/genproto v0.0.0-20191007204434-a023cd5227bd // indirect
+	google.golang.org/grpc v1.24.0 // indirect
```
Comment (kristofferahl) [inline go.mod]: Should indirect dependencies be included in go.mod? Asking because I don't know. :)

Comment (masutaka) [reply go.mod]: I may not understand `go get`. 4f079742a5ec7fa208390449ed03095fa205601d
I'm not sure, but it seems the `-u` option is not necessary. I will fix it later.

**Comment by masutaka** [comment]
Please review it again.

FYI: Many of the changes in go.sum are due to `go mod tidy`. It executed in `$ tf-sdk-migrator migrate`.

**Review by kristofferahl — APPROVED** [review]
Looks great @masutaka! Had to make some changes to the build in order to get it working (docker image layer caching is a paid feature it seems). Merging the PR!

**Comment by kristofferahl** [comment]
@masutaka Should we create a new release?

**Comment by masutaka** [comment]
Yes, it's good.

**Comment by kristofferahl** [comment]
Created a new release with the tag v1.4.0.
https://github.com/kristofferahl/terraform-provider-healthchecksio/releases/tag/v1.4.0
Great work!

**Pull request merged**